### ISSUE TITLE
Can't restart Shinken in GLPI

### ISSUE THREAD

**Issue by algorys** [ddurieux/glpi_monitoring#170]
Hello @ddurieux 

I've make some new install of Shinken and the different plugin in glpi (Monitoring and Web services) but when I try to restart shinken with GLPI, it won't.

I have all rights in GLPI for the plugin (Administration > Profile > Monitoring).
I have create web-service shinken and user shinken... in fact I follow the 0.84 plugin documentation...

I have the following issue : 
   - System state on monitoring webpage is red.
   - Can't restart shinken instance or with sql  database (ouput : you don't have rights to perform this action)

But the good news is that graphics are up and functionnal (with just some problems for vmware plugin perfdata)

Have you an idea to solve this ?

(maybe a relation with that : https://github.com/naparuba/shinken/issues/1629 )

Thanks in advance

GLPI 0.85.4
Shinken 2.4
Monitoring 0.85+1.1
Services Web 1.5.0

**Comment by algorys** [comment]
After delete a test catalog to see what it's happen, the plugin says : 

communication with shinken works good
L'ordre de rechargement Shinken a été envoyé à 2015-05-13 15:38
Shinken restarted at 2015-05-13 15:39

And I have see an error in the arbiter : 

[1431524337] EXTERNAL COMMAND: [1431524336] RELOAD_CONFIG
[1431524337] WARNING: [Shinken] RELOAD command : /etc/init.d/shinken reload
[1431524345] ERROR: [Shinken] [ws-arbiter] Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/shinken/basemodule.py", line 164, in start_module
    self._main()
  File "/usr/local/lib/python2.7/dist-packages/shinken/basemodule.py", line 296, in _main
    self.main()
  File "/var/lib/shinken/modules/ws-arbiter/module.py", line 407, in main
    inputready, _, _ = select.select(input, [], [], 1)
error: (4, 'Interrupted system call')

Maybe it help you ?

**Comment by mohierf** [comment]
On my configuration (Shinken 2.4 + Glpi 0.85 + Monitoring 0.85) I do not ahve any problem to restart from Glpi Monitoring interface.

 In my Shinken log:
```
[1431535065] WARNING: [Shinken] A command was received for service 'nsca_disk' on host 'kiosk-0001', but the service could not be found!
[1431535065] EXTERNAL COMMAND: [1431535064] PROCESS_SERVICE_CHECK_RESULT;kiosk-0001;nsca_disk;0;OK: |'C: used'=9GB;19;29;0;39 'C: used %'=24%;50;75;0;100
[1431535091] EXTERNAL COMMAND: [1431535091] RELOAD_CONFIG
[1431535091] WARNING: [Shinken] RELOAD command : nohup sh -c '/etc/init.d/shinken reload'  > /dev/null 2>&1 &
[1431535091]
[1431535096] INFO: [Shinken] [poller-master] We already got the conf 0 (scheduler-master)
[1431535096] INFO: [Shinken] [poller-master] Using max workers: 1
[1431535096] INFO: [Shinken] [poller-master] Using min workers: 1
```

 On my Shinken: 
```
## Module:      ws-arbiter
## Loaded by:   Arbiter, Receiver
# WebService module for the Arbiter and Receiver so you can send (POST) passive
# checks to it :)
define module {
    module_name     ws-arbiter
    module_type     ws_arbiter
    host            0.0.0.0
    port            7760
    username        anonymous   ; If you want auth, set username and password.
    #password        secret
}
```

 On my Glpi, I installed: php5-xmlrpc, php5-curl

 And then I configured Glpi as is : 

![image](https://cloud.githubusercontent.com/assets/2932687/7615760/09ab343c-f9a0-11e4-91a7-51841039bfc6.png)

![image](https://cloud.githubusercontent.com/assets/2932687/7615767/1c7504bc-f9a0-11e4-847f-6182a4b07480.png)


Hoping it helps ...

**Comment by algorys** [comment]
Thank you.

I try to make some little change (put user and password for webservice in shinken module and glpi module) but it still won't restart...

I don't understand why I have this difficulties...

The strangest thing is if I had or remove a catalogue or "composant" the order to restart is send...

**Comment by algorys** [comment]
Does I must have some specific right on the folder /var/www/glpi/plugins/monitoring ?

**Comment by algorys** [comment]
@ddurieux 
Hello David,

I've make the update GLPI 0.84 to 0.85 again (I've take a snapshot before) and so an update of the plugin Monitoring. And I have always the same issue : 

- System state on monitoring webpage is red.
- Can't restart shinken instance or with sql database (ouput : you don't have rights to perform this action)

If I put 127.0.0.1 or 192.168.x.x in the tag of Monitoring plugin it do the same. The plugins works but I can't restart from the link : 

Recharger la configuration de Shinken à partir de la base de données GLPI :	| Toutes les instances 
Redémarrer tous les démons Shinken :	| Toutes les instances 

It says always : Vous n'avez pas les droits requis pour réaliser cette action.

@mohierf says he has no this issue. Where I can make an error to have that ?

Thanks for help

**Comment by ddurieux** [comment]
for system state red, check if firewall on shinken server is opened for port 7770 and in arbiter.cfg, use address 0.0.0.0 instead localhost.

For the right, I have commited a fix today, oups yesterday now ^^

**Comment by algorys** [comment]
Fix system state with 0.0.0.0 in arbiter.cfg :)

I'm downloading the last commit to look the fix for right.
